magellan-library/src/main/java/com/wealthfront/magellan/navigation/Navigator.kt

```diff
@@ -4,10 +4,16 @@ import android.content.Context
 import android.view.View
 
 public interface NavigationRequestHandler {
-  public fun onNavigationRequested(
+  public fun shouldOverrideNavigation(
     navigationDelegate: NavigationDelegate,
     navigable: NavigableCompat
   ): Boolean
+
+  // what if this supplied a backstack operation, instead?
```

Comment:
TODO as noted I might change the API here. Currently there's an implicit contract that `overrideNavigationRequest` will call navigationDelegate.navigate somewhere in its code..we can make that explicit by having `overrideNavigationRequest` supply an alternative backstack operation, instead.